lib/straight-server/websocket_insight_client.rb

```diff
@@ -0,0 +1,56 @@
+module StraightServer
+  class WebsocketInsightClient
+
+    @@address_check_list = []
+    
+    class << self
+      
+      def start(url)
+        socket = SocketIO::Client::Simple.connect url
+
+        socket.on :connect do
+          StraightServer.logger.info "Connected to Insight websocket with url: #{url}"
+          socket.emit :subscribe, 'inv'
+          @wclient = StraightServer::WebsocketInsightClient.new
+        end
+
+        socket.on :tx do |data|
+          @wclient.check_transaction(data) if data["vout"]
+        end
+
+        socket.on :error do |err|
+          StraightServer.logger.warn err
+        end
+      end
+
+      def address_for_check_list
+        @@address_check_list
+      end
+
+      def add_address(address)
+        @@address_check_list.push(address).uniq!
+        StraightServer.logger.info "[WS] Added address for tracking: #{@@address_check_list}"
+      end
+
+      def remove_address(address)
+        @@address_check_list.delete(address)
+      end
+
+      def clear_address_list
+        @@address_check_list = []
+      end
+    end
+
+    def check_transaction(data)
+      data["vout"].each do |o|
+        if index = @@address_check_list.find_index(o.keys.first)
+          address = @@address_check_list.delete_at(index)
+          StraightServer.logger.info "[WS] Found transaction for address: #{address}"
+          StraightServer::Order.set_status_for(address, data)
+        end
+      end
+    end
+    
+  end
+
+end
```

Comment:
Well, that's quite a mess. It looks not thread-safe and higly coupled with the StraightServer::Order.
I suggest to:
1. not use class methods and class variables; instead just instantiate it during initialization and store in `StraightServer.insight_client`
2. provide an instance method `monitor(address, &block)`, which adds `address` to `@monitor` `Hash` as a key, and `block` as a value. When transaction detected, it calls that block with a `Struct` argument with all those fields https://github.com/MyceliumGear/straight-server/pull/6/files#diff-62c9cb9bdf687d36a4fa3c113e0af44eR70 (where they are assigned to Order, or whatever), and calls `stop_monitor(address)`, which deletes `address` key from `@monitor`.